Extensions/TextInput/JsExtension.js

```diff
@@ -0,0 +1,624 @@
+        .setType('color')
+        .setLabel(_('Border color'))
+        .setGroup(_('Field appearance'));
+
+      objectProperties
+        .getOrCreate('borderOpacity')
+        .setValue((objectContent.borderOpacity || 255).toString())
+        .setType('number')
+        .setLabel(_('Border opacity'))
+        .setGroup(_('Field appearance'));
+
+      return objectProperties;
+    };
+    textInputObject.setRawJSONContent(
+      JSON.stringify({
+        initialValue: '',
+        placeholder: 'Touch to start typing',
+        fontResourceName: '',
+        fontSize: 20,
+        inputType: 'text',
+        textColor: '0;0;0',
+        fillColor: '255;255;255',
+        fillOpacity: 1,
+        borderColor: '0;0;0',
+        borderOpacity: 1,
+      })
+    );
+
+    // $FlowExpectedError - ignore Flow warning as we're creating an object
+    textInputObject.updateInitialInstanceProperty = function (
+      objectContent,
+      instance,
+      propertyName,
+      newValue,
+      project,
+      layout
+    ) {
+      if (propertyName === 'initialValue') {
+        instance.setRawStringProperty('initialValue', newValue);
+        return true;
+      } else if (propertyName === 'placeholder') {
+        instance.setRawStringProperty('placeholder', newValue);
+        return true;
+      }
+
+      return false;
+    };
+    // $FlowExpectedError - ignore Flow warning as we're creating an object
+    textInputObject.getInitialInstanceProperties = function (
+      content,
+      instance,
+      project,
+      layout
+    ) {
+      const instanceProperties = new gd.MapStringPropertyDescriptor();
+
+      instanceProperties
+        .getOrCreate('initialValue')
+        .setValue(instance.getRawStringProperty('initialValue'))
+        .setType('string')
+        .setLabel(_('Initial value'));
+      instanceProperties
+        .getOrCreate('placeholder')
+        .setValue(instance.getRawStringProperty('placeholder'))
+        .setType('string')
+        .setLabel(_('Placeholder'));
+
+      return instanceProperties;
+    };
+
+    const object = extension
+      .addObject(
+        'TextInputObject',
+        _('Text input'),
+        _('A text field the player can type text into.'),
+        'JsPlatform/Extensions/text_input.svg',
+        textInputObject
+      )
+      .setIncludeFile('Extensions/TextInput/textinputruntimeobject.js')
+      .addIncludeFile(
+        'Extensions/TextInput/textinputruntimeobject-pixi-renderer.js'
+      );
+
+    // Properties expressions/conditions/actions:
+    object
+      .addExpressionAndConditionAndAction(
+        'string',
+        'Text',
+        _('Text'),
+        _('the text'),
+        _('the text'),
+        '',
+        'res/conditions/text24.png'
+      )
+      .addParameter('object', _('Text input'), 'TextInputObject', false)
+      .useStandardParameters('string')
+      .setFunctionName('setValue')
+      .setGetter('getValue');
```

Comment:
"Text" might be more specific than "String".

Reply:
I'd still use String for consistency (see Variables and the current text entry object). Since this is a javascript API anyways, and string refers to text in JavaScript, I think this is fine.

Reply:
Changed to get/setString for consistency with other objects.